src/test/scala/org/constellation/cluster/SnapshotRollbackTest.scala

```diff
@@ -0,0 +1,70 @@
+package org.constellation.cluster
+
+import better.files.File
+import cats.implicits._
+import org.constellation.consensus.{Snapshot, SnapshotInfo, StoredSnapshot}
+import org.constellation.primitives.Transaction
+import org.constellation.serializer.KryoSerializer
+import org.scalatest.{FlatSpec, Ignore}
+
+class SnapshotRollbackTest extends FlatSpec {
+
+  private val snapshotsFolder = "rollback_data"
+  private val snapshotInfoFile = "rollback_info"
+  private val zeroSnapshotHash = Snapshot("", Seq()).hash
+
+  type AccountBalances = Map[String, Long]
+
+  "Rollback" should "validate snapshots from files" in {
+    if (File("rollback_data").notExists) cancel
+
+    val snapshots = loadSnapshotsFromFile()
+    val snapshotInfo = loadSnapshotInfoFromFile()
+
+    val newestSnapshot = findSnapshot(snapshotInfo.snapshot.lastSnapshot, snapshots)
+    val accountBalances = calculateAccountBalances(newestSnapshot.snapshot.hash, snapshots)
+
+    validateAccountBalance(accountBalances)
```

Comment:
This test doesn't check if the rollback validates snapshots but checks if your `validateAccountBalance` function from tests works properly